src/Flysystem/FlysystemAssetStore.php

```diff
@@ -597,10 +598,16 @@ function (ParsedFileID $parsedFileID, Filesystem $fs, FileResolutionStrategy $st
                             // Move cached hash value to new location
                             $hasher->move($origin, $fs, $destination);
                         }
-                        $this->truncateDirectory(dirname($origin ?? ''), $fs);
+                        $tryTruncate[dirname($origin)] = true;
```

Comment:
```suggestion
                        $dirsToTruncate[] = dirname($origin);
```

Cleaner to just do this an use array_unique() later. There won't be many dirs to truncate so the performance costs is minuscule and won't be noticed by end user

Repeat this for all examples below

Reply:
I don't see how it's cleaner - but it's about same/same so done.